### PR TITLE
[Feature] [Inference]Add ROCm platform support for single-gpu inference

### PR DESCRIPTION
**Pull request by sopiko99**
## Description
This PR adds support for **single-GPU inference** on **ROCm** platform. 
Supported attention backends: `FLASH_ATTN`, `TORCH_SDPA`

⚠️Note: Currently not supported distributed inference
⚠️Note: Currently not supported attention backends `SLIDING_TILE_ATTN`, `SAGE_ATTN`.
## Hardware
**AMD Instinct™ GPUs** : This was tested on an AMD Instinct MI300X GPU. Ensure you are using an AMD Instinct GPU or compatible hardware with ROCm support and that your system meets the [official requirements](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html).

## Software
* **ROCm 6.3** : Install and verify ROCm by following the [ROCm install guide](https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/quick-start.html). After installation, confirm your setup using:
  ```
  rocm-smi
  ```
  This command lists your AMD GPUs with relevant details.
* **Docker** : Ensure Docker is installed and configured correctly. Follow the Docker installation guide for your operating system.

## Launch the Docker Container
Pull the Docker image required for this PR: `docker pull rocm/pytorch-training:v25.6`
Launch the Docker container in detach mode and map the necessary directories.

```bash
docker run -d \
  --network=host \
  --device=/dev/kfd \
  --device=/dev/dri \
  --group-add=video \
  --ipc=host \
  --cap-add=SYS_PTRACE \
  --security-opt seccomp=unconfined \
  --privileged \
  --name fastvideo-tmp \
  -v $(pwd):/workspace/ \
  rocm/pytorch-training:v25.6 \
  tail -f /dev/null
```
You can **enter the container interactively** with:  `docker exec -it fastvideo-tmp bash`
## Install Dependencies

Update in **pyproject.toml** file variable `dependencies` values with this:
```
dependencies = [
    # Core Libraries
    "scipy==1.14.1", "six==1.16.0", "h5py==3.12.1",

    # Machine Learning & Transformers
    "timm==1.0.11", "peft>=0.15.0", "diffusers>=0.33.1",

    # Computer Vision & Image Processing
    "opencv-python==4.10.0.84", "pillow>=10.3.0", "imageio==2.36.0",
    "imageio-ffmpeg==0.5.1", "einops",

    # Experiment Tracking & Logging
    "wandb>=0.19.11", "loguru", "test-tube==0.7.5",

    # Miscellaneous Utilities
    "tqdm", "pytest", "PyYAML==6.0.1", "protobuf>=5.28.3",
    "gradio>=5.22.0", "moviepy==1.0.3", "flask",
    "flask_restful", "aiohttp", "huggingface_hub", "cloudpickle",
    # System & Monitoring Tools
    "gpustat", "watch", "remote-pdb",

    # Kernel & Packaging
    "wheel",

    # Training Dependencies
    "torchdata",
    "pyarrow",
    "datasets",
    "av",
]
```

And then run install:
```
pip install -e .
```

### bitsandbytes (Optional)
This dependency has been removed recently, but in case it's needed this is how to install it on rocm:
```bash
git clone --recurse https://github.com/ROCm/bitsandbytes 
cd bitsandbytes
git checkout rocm_enabled # It should be the default branch, but make sure that we are there
cmake -DCOMPUTE_BACKEND=hip -S . -DBNB_ROCM_ARCH="gfx942" # MI300x arch
make
pip install .
```

## Run Inference
Create a Python file `generate_video.py` and copy the following:

```python
from fastvideo import VideoGenerator


def main():
    # Create a video generator with a pre-trained model
    generator = VideoGenerator.from_pretrained(
        "Wan-AI/Wan2.1-T2V-1.3B-Diffusers",
        num_gpus=1,  # Currently only supports 1 GPU inference on rocm platform
    )
    prompt = "Red panda playing in a snowy forest, surrounded by pine trees and falling snowflakes."
  
    # With TeaCache
    video = generator.generate_video(
        prompt,
        return_frames=True,  # Also return frames from this call (defaults to False)
        output_path="my_videos/",  # Controls where videos are saved
        save_video=True,
        enable_teacache=True  # Enable TeaCache for caching
    )

if __name__ == '__main__':
    main()
```

#### Flash Attention 2 and 3
`FASTVIDEO_ATTENTION_BACKEND=FLASH_ATTN python generate_video.py`
#### Torch SDPA
`FASTVIDEO_ATTENTION_BACKEND=TORCH_SDPA python generate_video.py`

### CODE REVIEW

**Comment by jzhang38** [comment]
Thanks for your contritbution. We will try to get this merged soon.

**Review by SolitaryThinker — APPROVED** [review]
Thank you for your contributions! The CI test failures are unrelated.
